### DISCUSSION TITLE
Distributed Agent Runtime: Adding secure channel communication for worker runtimes

### DISCUSSION THREAD

**Discussion by prabhakaran-annadurai** [microsoft/autogen#4101] [Ideas]
version: 0.4

Scenario:

Host service is running in an Azure Container Apps and worker runtime running in a different service like Azure functions & trying to connect to Host.

Current behavior: worker runtime is not able to connect to host as secure channel is not implemented (only grpc.aio.insecure_channel communication is implemented in python/packages/autogen-core/src/autogen_core/application/_worker_runtime.py ).

Expected: worker runtime must be able to communicate with host using a secure channel with grpc.ssl_channel_credentials() when non-local host is provided.

**Comment by ekzhu** [comment]
Absolutely, this is on our roadmap https://github.com/microsoft/autogen/issues/4103
